editor/connections_dialog.cpp

```diff
@@ -244,14 +330,50 @@ void ConnectDialog::_notification(int p_what) {
+			type_icons[Variant::INT] = get_theme_icon("int", "EditorIcons");
+			type_icons[Variant::FLOAT] = get_theme_icon("float", "EditorIcons");
+			type_icons[Variant::STRING] = get_theme_icon("String", "EditorIcons");
+			type_icons[Variant::VECTOR2] = get_theme_icon("Vector2", "EditorIcons");
+			type_icons[Variant::VECTOR2I] = get_theme_icon("Vector2i", "EditorIcons");
+			type_icons[Variant::RECT2] = get_theme_icon("Rect2", "EditorIcons");
+			type_icons[Variant::RECT2I] = get_theme_icon("Rect2i", "EditorIcons");
+			type_icons[Variant::VECTOR3] = get_theme_icon("Vector3", "EditorIcons");
+			type_icons[Variant::VECTOR3I] = get_theme_icon("Vector3i", "EditorIcons");
+			type_icons[Variant::TRANSFORM2D] = get_theme_icon("Transform2D", "EditorIcons");
+			type_icons[Variant::VECTOR4] = get_theme_icon("Vector4", "EditorIcons");
+			type_icons[Variant::VECTOR4I] = get_theme_icon("Vector4i", "EditorIcons");
+			type_icons[Variant::PLANE] = get_theme_icon("Plane", "EditorIcons");
+			type_icons[Variant::QUATERNION] = get_theme_icon("Quaternion", "EditorIcons");
+			type_icons[Variant::AABB] = get_theme_icon("AABB", "EditorIcons");
+			type_icons[Variant::BASIS] = get_theme_icon("Basis", "EditorIcons");
+			type_icons[Variant::TRANSFORM3D] = get_theme_icon("Transform3D", "EditorIcons");
+			type_icons[Variant::PROJECTION] = get_theme_icon("Projection", "EditorIcons");
+			type_icons[Variant::COLOR] = get_theme_icon("Color", "EditorIcons");
+			type_icons[Variant::STRING_NAME] = get_theme_icon("StringName", "EditorIcons");
+			type_icons[Variant::NODE_PATH] = get_theme_icon("NodePath", "EditorIcons");
+			type_icons[Variant::RID] = get_theme_icon("RID", "EditorIcons");
+			type_icons[Variant::OBJECT] = get_theme_icon("Object", "EditorIcons");
+			type_icons[Variant::CALLABLE] = get_theme_icon("Callable", "EditorIcons");
+			type_icons[Variant::SIGNAL] = get_theme_icon("Signal", "EditorIcons");
+			type_icons[Variant::DICTIONARY] = get_theme_icon("Dictionary", "EditorIcons");
+			type_icons[Variant::ARRAY] = get_theme_icon("Array", "EditorIcons");
+			type_icons[Variant::PACKED_BYTE_ARRAY] = get_theme_icon("PackedByteArray", "EditorIcons");
+			type_icons[Variant::PACKED_INT32_ARRAY] = get_theme_icon("PackedInt32Array", "EditorIcons");
+			type_icons[Variant::PACKED_INT64_ARRAY] = get_theme_icon("PackedInt64Array", "EditorIcons");
+			type_icons[Variant::PACKED_FLOAT32_ARRAY] = get_theme_icon("PackedFloat32Array", "EditorIcons");
+			type_icons[Variant::PACKED_FLOAT64_ARRAY] = get_theme_icon("PackedFloat64Array", "EditorIcons");
+			type_icons[Variant::PACKED_STRING_ARRAY] = get_theme_icon("PackedStringArray", "EditorIcons");
+			type_icons[Variant::PACKED_VECTOR2_ARRAY] = get_theme_icon("PackedVector2Array", "EditorIcons");
+			type_icons[Variant::PACKED_VECTOR3_ARRAY] = get_theme_icon("PackedVector3Array", "EditorIcons");
+			type_icons[Variant::PACKED_COLOR_ARRAY] = get_theme_icon("PackedColorArray", "EditorIcons");
```

Comment:
Not happy with this. I think we could have some generic method e.g. in EditorNode. It would help #21916 too.

Reply:
just `get_theme_icon(Variant::get_type_name( variant.get_type() ), "EditorIcons")` is enough.